tiledb/bioimg/converters/base.py

```diff
@@ -58,80 +60,64 @@
 from .tiles import iter_tiles, num_tiles
 
 
-class ImageReader(ABC):
-    @abstractmethod
-    def __init__(
-        self,
-        input_path: str,
-        *,
-        logger: Optional[logging.Logger],
-        source_config: Optional[tiledb.Config] = None,
-        source_ctx: Optional[tiledb.Ctx] = None,
-        dest_config: Optional[tiledb.Config] = None,
-        dest_ctx: Optional[tiledb.Config] = None,
-        **kwargs: Any,
-    ):
-        """Initialize this ImageReader"""
+class ImageReader(Protocol):
 
-    def __enter__(self) -> ImageReader:
-        return self
+    def __enter__(self) -> ImageReader: ...
 
-    def __exit__(self, exc_type: Any, exc_val: Any, exc_tb: Any) -> None:
-        pass
+    def __exit__(self, exc_type: Any, exc_val: Any, exc_tb: Any) -> None: ...
 
     @property
-    @abstractmethod
     def source_ctx(self) -> tiledb.Ctx:
         """The ctx of the source path of this image reader."""
+        ...
 
     @property
-    @abstractmethod
     def dest_ctx(self) -> tiledb.Ctx:
         """The ctx of the dest path of this image reader."""
+        ...
 
     @property
-    @abstractmethod
     def logger(self) -> Optional[logging.Logger]:
         """The logger of this image reader."""
+        ...
 
     @logger.setter
-    @abstractmethod
     def logger(self, default_logger: logging.Logger) -> None:
         """The setter for the logger of this image reader"""
+        ...
```

Comment:
Since this is gettable/settable, you could [make this a simple property](https://typing.readthedocs.io/en/latest/spec/protocol.html#protocols:~:text=class%20Template(Protocol)%3A%0A%20%20%20%20name%3A%20str%20%20%20%20%20%20%20%20%23%20This%20is%20a%20protocol%20member%0A%20%20%20%20value%3A%20int%20%3D%200%20%20%20%23%20This%20one%20too%20(with%20default)):

```python
class ImageReader(Protocol):
  ...
  logger: Optional[logging.Logger]
  ...
```